instruqt/protecting-your-assets/04-helm-values-for-license-fields/assignment.md

```diff
@@ -0,0 +1,273 @@
+
+Available License Values
+========================
+
+The global values provided by the Replicated registry are the most valuable to
+use within your Helm chart. Let's take a look at our Slackernews license to see
+what other information about the license is available. Log in to the Replicated
+registry as your Nitflex customer.
+
+```
+helm registry login [[ Instruqt-Var key="REGISTRY_HOST" hostname="shell" ]]  --username [[ Instruqt-Var key="REGISTRY_USERNAME" hostname="shell" ]]  --password [[ Instruqt-Var key="REGISTRY_PASSWORD" hostname="shell" ]]
+```
+
+You can then view the values using the `helm show values` command. The command
+below will get the values and isolate the globals. For Slackernews, this
+will only include the values injected by the Registry.
+
+```
+helm show values oci://[[ Instruqt-Var key="REGISTRY_HOST" hostname="shell"]]/[[ Instruqt-Var key="REPLICATED_APP" hostname="shell" ]]/slackernews | yq -P .global
+```
+
+You'll see something like (the encoded fields and expiration will probably be
+different).
+
+```
+replicated:
+  channelName: Unstable
+  customerEmail: [[ Instruqt-Var key="CUSTOMER_EMAIL" hostname="shell" ]]
+  customerName: Nitflex
+  dockerconfigjson: eyJhdXRocyI6eyJpbWFnZXMuc2hvcnRyaWIuaW8iOnsiYXV0aCI6Ik1tWlhSMVEzZDBGck0yOVVTbGhWY25GVWFrWlFUV0k1ZVV4dE9qSm1WMGRVTjNkQmF6TnZWRXBZVlhKeFZHcEdVRTFpT1hsTWJRPT0ifSwicmVnaXN0cnkuc2hvcnRyaWIuaW8iOnsiYXV0aCI6Ik1tWlhSMVEzZDBGck0yOVVTbGhWY25GVWFrWlFUV0k1ZVV4dE9qSm1WMGRVTjNkQmF6TnZWRXBZVlhKeFZHcEdVRTFpT1hsTWJRPT0ifX19
+  licenseFields:
+    expires_at:
+      name: expires_at
+      title: Expiration
+      description: License Expiration
+      value: "2025-05-31T00:00:00Z"
+      valueType: String
+      signature:
+        v1: UwI4/IL6JR4K5Tw7gapvfW6+zkirfMulbAxaQqkVIAZazip+pehegNRVhHEhbM9V9EibONGBbOazipb8aQeWO2hYoN0mcQOelUxVmK7U2GFP862tyorwAPwxMg+ZbAunUsoKP4/GT+Up5bhC8UN+NgyfZFzmCo3G6TK+2tbtI/tHXN0IwFacY3TyvryAfB+6qRGpsb0efb/Wl4DNmzuo/z9qE/1HbdG/TGUdq3SEmaH4iGBSeUZMHrHmW7/fHM8DDAABi7NW8v7HGEKI467yufKPcohzim8roLl2mdLsnLq2o3J5ovsAImgXUK3ac7ymHBLdT9WRg5cuBtAWNI5oaQ==
+  licenseID: 2fWGT7wAk3oTJXUrqTjFPMb9yLm
+  licenseType: prod
+```
+
+There are a few useful things in there:
+
+* Which customer this license is for, including name, a unique license ID, and
+  their email.
+* The customer's credentials for the Replicated registry and proxy (we used
+  this earlier).
+* The type of license: `prod` (representing a paid license), `dev`, `trial`, or
+  `community`.
+* A set of fields representing license entitlements. We'll talk about those in
+  the next section.
+
+You can use this information to customize the user experience, guard components
+(e.g. don't allow persistence with trials), and tailor the install to the
+license entitlements.
+
+License Fields
+==============
+
+The license has several fields that represent specific entitlements of this
+license. In Helm, they're available as an array
```

Comment:
```suggestion
license. In Helm, they're available as a dictionary
```

Looks like a Dict to me

Reply:
You're right. I was thinking they were an array of dicts when I wrote it.